### PR TITLE
[Ruby][Faraday][#10137] Fix config request options and timeout

### PR DESCRIPTION
**Pull request by NivathaSV8**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by NivathaSV8** [comment]
@wing328 Can you please review this PR and get it merged.

**Comment by wing328** [comment]
cc @cliffano (2017/07) @zlx (2017/09) @autopp (2019/02)

**Comment by wing328** [comment]
Ruby tests with updated samples passed via https://app.travis-ci.com/github/OpenAPITools/openapi-generator/builds/238968179.

I'll update samples after merging this PR into master.

File: modules/openapi-generator/src/main/resources/ruby-client/api_client_faraday_partial.mustache [hunk 1]
```diff
@@ -73,23 +73,20 @@
...
       if [:post, :patch, :put, :delete].include?(http_method)
         req_body = build_request_body(header_params, form_params, opts[:body])
-        req_opts.update :body => req_body
         if @config.debugging
           @config.logger.debug "HTTP request body param ~BEGIN~\n#{req_body}\n~END~\n"
         end
       end
       request.headers = header_params
       request.body = req_body
+      request.options = OpenStruct.new(req_opts)
```
Comment (ckoegel) [inline modules/openapi-generator/src/main/resources/ruby-client/api_client_faraday_partial.mustache:89]: @wing328 @NivathaSV8 I've been attempting to use this feature in a generated ruby SDK, as well as add the ability to directly return the binary media data gotten for a file. It appears to me that this PR has broken the functionality of the code fixed in #7842. While the temp file is created, the file is empty, and directly returning the binary data shows that `@stream` is empty. Prior to this PR, this seemed to work properly, and the culprit appears to be this line (89), as removing it allows the temp file to be created properly and the binary media return I'm working on in #11254 to work as well. I'm still fairly new to ruby, so I may be missing something here, but is there any way this issue could be validated? I've generated the SDK with version 5.3.1 of the CLI that was current as of 1/7/2021. I'm happy to provide more of my findings about this issue to help fix it, but I'd like to make sure I'm not incorrectly using this feature first. Any explanation as to why I might be seeing this issue and what could be done to fix it is greatly appreciated, thanks!

**Pull request merged**